src/components/FileUploadField/FileUploadField.tsx

```diff
@@ -0,0 +1,81 @@
+import { makeStyles } from "@material-ui/core";
+import Button from "@material-ui/core/Button";
+import IconButton from "@material-ui/core/IconButton";
+import DeleteIcon from "@material-ui/icons/Delete";
+import { commonMessages } from "@saleor/intl";
+import React from "react";
+import { useIntl } from "react-intl";
+
+export interface FileUploadFieldProps
+  extends React.DetailedHTMLProps<
+    React.InputHTMLAttributes<HTMLInputElement>,
+    HTMLInputElement
+  > {
+  disabled: boolean;
+  fileName?: string;
+  onFileUpload: (file: File) => void;
+  onFileDelete: () => void;
+}
+
+const useStyles = makeStyles(
+  theme => ({
+    fileField: {
+      display: "none"
+    },
+    uploadFileContent: {
+      color: theme.palette.primary.main,
+      fontSize: "1rem"
+    }
+  }),
+  { name: "FileUploadField" }
+);
+
+const FileUploadField: React.FC<FileUploadFieldProps> = props => {
+  const { disabled, fileName, onFileUpload, onFileDelete, ...rest } = props;
```

Comment:
Maybe input props should be separate prop?

Reply:
sure, good idea

Reply:
updated